content/docs/guides/crosswalk/kubernetes/aws/02-kubernetes-stacks/01-cluster-configuration.md

```diff
@@ -0,0 +1,672 @@
+
+**Networking**
+
+  * Public subnets for use with public load balancers.
+  * Private subnets for use with private load balancers.
+  * Private subnets used as the default subnets for workers, by design.
+  * Pod networking managed by the [AWS CNI Plugin][aws-k8s-cni].
+
+**Storage**
+
+  * Default Kubernetes storage classes backed by cloud volume types.
+
+**Workers**
+
+  * No default worker node group in favor of specific, separately managed node groups.
+
+**General**
+
+  * A specific version of Kubernetes for the control plane.
+  * Tags for resources under cluster management.
+  * Control plane logging enabled.
+  * Configurable accessibility for the cluster API endpoint (optional).
+
+    ```typescript
+    import * as eks from "@pulumi/eks";
+    
+    // Create an EKS cluster.
+    const cluster = new eks.Cluster(`${projectName}`, {
+        instanceRoles: [
+            aws.iam.Role.get("adminsIamRole", stdNodegroupIamRoleName),
+            aws.iam.Role.get("devsIamRole", perfNodegroupIamRoleName),
+        ],
+        roleMappings: [
+            {
+                roleArn: config.adminsIamRoleArn,
+                groups: ["system:masters"],
+                username: "pulumi:admins",
+            },
+            {
+                roleArn: config.devsIamRoleArn,
+                groups: ["pulumi:devs"],
+                username: "pulumi:alice",
+            },
+        ],
+        vpcId: config.vpcId,
+        publicSubnetIds: config.publicSubnetIds,
+        privateSubnetIds: config.privateSubnetIds,
+        storageClasses: {
+            "gp2": { type: "gp2", encrypted: true},
+            "sc1": { type: "sc1"}
+        },
+        nodeAssociatePublicIpAddress: false,
+        skipDefaultNodeGroup: true,
+        deployDashboard: false,
+        version: "1.14",
+        tags: {
+            "Project": "k8s-aws-cluster",
+            "Org": "pulumi",
+        },
+        clusterSecurityGroupTags: { "ClusterSecurityGroupTag": "true" },
+        nodeSecurityGroupTags: { "NodeSecurityGroupTag": "true" },
+        enabledClusterLogTypes: ["api", "audit", "authenticator", "controllerManager", "scheduler"],
+        // endpointPublicAccess: false,     // Requires bastion to access cluster API endpoint
+        // endpointPrivateAccess: true,     // Requires bastion to access cluster API endpoint
+    });
+    
+    // Export the cluster details.
+    export const kubeconfig = cluster.kubeconfig;
+    export const clusterName = cluster.core.cluster.name;
+    ```
+
+## Create the Worker Nodes
+
+Create new worker node groups equipped with:
+
+**Identity**
+
+  * IAM role instance profiles for each worker node group.
+
+**Networking**
+
+  * No public IP addresses assigned to node group workers.
+  * Configurable security groups and ingress rules.
+
+**Workers**
+
+  * An AMI for a specific version of Kubernetes to use against the control plane.
+  * Separate machine types by use-case.
```

Comment:
This is definitely subtle. This feels like it warrants at least a paragraph or two of explanation, along with how it relates to labels and taints as mentioned in passing below.

Reply:
Done